core/chains/evm/logpoller/dsl.go

```diff
@@ -21,7 +38,18 @@ func (f *EventTopicsByValueFilter) Accept(visitor commontypes.Visitor) {
 }
 
 type FinalityFilter struct {
-	confs Confirmations
+	Confs Confirmations
+}
+
+func NewFinalityFilter(filter *commontypes.ConfirmationFilter) (*FinalityFilter, error) {
+	switch filter.Confirmations {
+	case commontypes.Finalized:
+		return &FinalityFilter{Finalized}, nil
+	case commontypes.Unconfirmed:
+		return &FinalityFilter{Unconfirmed}, nil
```

Comment:
Maybe add a TODO here for supporting partial confirmations (confidence levels) like CCIP has asked for?

Reply:
added a TODO